### PR TITLE
Suppress caching warning when compiling Numba functions

### LINKED ISSUE #669: Suppress NumbaWarning
I get this all the time, not useful: 

```
NumbaWarning: Cannot cache compiled function "numba_funcified_fgraph" as it uses dynamic globals (such as ctypes pointers and large global arrays)
```

### PR DESCRIPTION
**Pull request by ricardoV94**
## Related Issue

- [x] Closes #669 

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [x] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

File: pytensor/link/numba/dispatch/basic.py [hunk 1]
```diff
@@ -61,6 +61,13 @@ def global_numba_func(func):
 def numba_njit(*args, **kwargs):
     kwargs.setdefault("cache", config.numba__cache)
 
+    # Supress caching warnings
```
Comment (aseyboldt) [inline pytensor/link/numba/dispatch/basic.py:64]: Maybe we should restore the warnings after compilation so that this doesn't impact unrelated code? (https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings)
I guess in that case we actually need the same thing in nutpie...

Comment (ricardoV94) [reply pytensor/link/numba/dispatch/basic.py:64]: I tried that but it required a wrapper around the function returned by this function, as it is triggered in at least two places: evaluating the function on VM and when dispatching on Composite. I was also worried of the overhead of entering the context and defining the filter every time we call the function on the VM

Comment (aseyboldt) [reply pytensor/link/numba/dispatch/basic.py:64]: Ah, I see...
In that case, can we add `numba_funcified_fgraph` to the pattern? In that case it still should only apply to our own function.

Comment (ricardoV94) [reply pytensor/link/numba/dispatch/basic.py:64]: Yes we can do that

**Review by aseyboldt — APPROVED** [review]
Looks good :-)
